feat: adding recipes tracker workshop (#56456)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-recipe-tracker/66fbcf750a62784cf11f5631.md

---
id: 66fbcf750a62784cf11f5631
title: Step 7
challengeType: 1
dashedName: step-7
---

# --description--

> Now you should work on calculating the `averageRating`, `totalIngredients`, and the `difficultyLevel` for each recipe in the `recipes` array.

Reviewer: ```suggestion
Now, you should work on calculating the `averageRating`, `totalIngredients`, and the `difficultyLevel` for each recipe in the `recipes` array.
```

> Start by creating a `getAverageRating` function with a `ratings` parameter. Inside the function, calculate the average rating by making use of the items in the `ratings` property of each recipe.

Reviewer: ```suggestion
Start by creating a `getAverageRating` function that takes a `ratings` parameter. Inside the function, calculate the average rating using the items in the `ratings` property of each recipe.
```

---
id: 66fbcf750a62784cf11f5631
title: Step 7
challengeType: 1
dashedName: step-7
---

# --description--

Now you should work on calculating the `averageRating`, `totalIngredients`, and the `difficultyLevel` for each resipe in the `recipes` array.

Start by creating a `getAverageRating` function with a `ratings` parameter. Inside the function, calculate the average rating by making use of the items in the `ratings` property of each recipe.

Your `getAverageRating` function must return a number.

# --hints--

You should create a `getAverageRating` function.

```js
assert.isFunction(getAverageRating)
```

You `getAverageRating` function should have a `ratings` parameter.

```js
assert.match(getAverageRating.toString(), /ratings/);
```

> You `getAverageRating` function should return a number.

Reviewer 2: There should be another test here to prevent campers from hardcoding a result like this
<img width="738" alt="Screenshot 2024-10-10 at 5 06 43 PM" src="https://github.com/user-attachments/assets/a7ded3e9-e152-415f-add9-bfe05b3f4fa6">

I would suggest a test that calls the function a few tests with the other recipe objects to prevent hardcoding

Author: Can you check again to see if what I have is good enough?
